src/main/java/com/ericsson/eiffel/remrem/semantics/validator/EiffelValidator.java

```diff
@@ -72,41 +89,127 @@ public EiffelValidator(String schemaResourceName, String eventType, List<String>
+        JsonParser parser = new JsonParser();
+        DocumentContext doc = JsonPath.parse(jsonObjectInput.toString());
+        for (ProcessingMessage processingMessage : report) {
+            if (LogLevel.ERROR.equals(processingMessage.getLogLevel())) {
+                JsonElement element = parser.parse(processingMessage.asJson().toString());
+                if (element.getAsJsonObject().get(KEYWORD).getAsString().equals(REQUIRED)) {
+                    throw new EiffelValidationException(getErrorsList(report));
+                }
+                String errorPath = element.getAsJsonObject().get(INSTANCE).getAsJsonObject().get(POINTER)
+                        .getAsString();
+                JsonObject addValidationFailurs = addValidationFailurs(element, processingMessage.getMessage());
+                remremGenerateFailures.add(addValidationFailurs);
+                doc.delete(getPath(errorPath));
+            }
+        }
+        return doc.jsonString();
+    }
+    
+    /**
+     * getPath for create the DocumentContext path from validation failure path
+     * example 
+     * 1) meta/tags/0 --> $.meta.tags[0]
+     * 2) links/0/target --> $.links[0]
+     * 3) /data/outcome/conclusion --> $.data.outcome.conclusion
+     * @param errorPath
+     * @return path
+     */
+    private String getPath(String errorPath) {
+        final Pattern pattern = Pattern.compile(REGEX_PATH, Pattern.MULTILINE);
+        final Matcher matcher = pattern.matcher(errorPath);
+        while (matcher.find()) {
+            if(matcher.group(0).length()>1) {
+                errorPath = errorPath.substring(0, errorPath.indexOf(matcher.group(0))+matcher.group(0).length());
+                errorPath = errorPath.replaceAll(matcher.group(0), "["+matcher.group(0).substring(1)+"]");
+            }
+        }
+        return "$" + errorPath.replace(SLASH, DOT);
+    }
+    private JsonObject addValidationFailurs(JsonElement element, String message) {
+        log.debug("Adding the error field information to the array");
+        String type = element.getAsJsonObject().get(KEYWORD).getAsString();
+        String path = element.getAsJsonObject().get(INSTANCE).getAsJsonObject().get(POINTER).getAsString();
+        JsonObject object = new JsonObject();
+        object.addProperty(TYPE2, type);
+        object.addProperty(MESSAGE2, message);
+        object.addProperty(PATH2, path);
+        return object;
+    }
+    private JsonObject addRemremGenerateFailuresToCustomData(JsonObject inputJson, JsonArray remremGenerateFailures) {
+        JsonArray customData = getCustomData(inputJson);
+        JsonObject object = new JsonObject();
+        object.add(REMREM_GENERATE_FAILURES, remremGenerateFailures);
+        customData.add(object);
+        return inputJson;
+    }
+    
+    /**
+     * for get the customData json array from Gen2 Eiffel message.
+     * @param json
+     * @return
```

Comment:
* A better wording of the description would be "Gets the customData array from an Eiffel message" (dropping the "Gen2" comment which seems like an Ericsson-ism).
* The empty `@param` and `@return` lines are just noise.